### PR TITLE
Updates, problems panel bugfix

### LINKED ISSUE #703: Problems panel starts open even if it's empty
it shouldn't do dat

### PR DESCRIPTION
**Pull request by ethteck**
fixes #703

### CODE REVIEW

**Comment by ethteck** [comment]
fixes #703

File: frontend/src/lib/device.ts [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
 
 export function getUserAgent(): string {
     if (isServer()) {
-        try {
-            return headers().get("user-agent")
-        } catch (error) {
-            console.warn("Failed to get user-agent from headers:", error)
-            return ""
-        }
+        console.error("getUserAgent() called on server")
```
Comment (nanaian) [inline frontend/src/lib/device.ts:9]: maybe throw

File: frontend/src/pages/_app.tsx [hunk 2]
```diff
@@ -56,12 +55,6 @@ export default function MyApp({ Component, pageProps }) {
         document.body.style.setProperty("--code-line-height", codeLineHeight)
     }, [codeLineHeight])
 
-    useEffect(() => {
-        if (isMacOS()) {
-            document.body.classList.add("device-macos")
-        }
-    }, [])
```
Comment (nanaian) [inline frontend/src/pages/_app.tsx:63]: hm why?

Comment (ethteck) [reply frontend/src/pages/_app.tsx:63]: this class has since been removed from our css, and I wanted to clean up dead code

**Pull request merged**